CSharpSource/Source/api/Presence/PresenceWriter.cs

```diff
@@ -0,0 +1,169 @@
+﻿using Microsoft.Xbox.Services.Shared.Presence;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Xbox.Services.Presence
+{
+    public class PresenceWriter: IPresenceWriter
+    {
+        private const string PresenceWriterApiContract = "3";
+        private static readonly Uri UserPresenceBaseUri = new Uri("https://userpresence.xboxlive.com");
+        private static readonly string HeartBeatFromServiceHeaderName = "X-Heartbeat-After";
+
+        private static IPresenceWriter instance;
+        private static readonly object instanceLock = new object();
+
+        private const int defaultHeartBeatDelayInMins = 5;
+        private bool writerInProgress;
+        private bool isCallInProgess;
+        private bool stopTimerCalled;
+        private Timer timer;
+        private Dictionary<XboxLiveUser, UserPresenceWriterStatus> usersBeingTrackedMap;
+
+        private PresenceWriter()
+        {
+            writerInProgress = false;
+            isCallInProgess = false;
+            usersBeingTrackedMap = new Dictionary<XboxLiveUser, UserPresenceWriterStatus>();
+        }
+
+        internal static IPresenceWriter Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    lock (instanceLock)
+                    {
+                        if (instance == null)
+                        {
+                            instance = XboxLive.UseMockServices ? new MockPresenceWriter() : (IPresenceWriter) new PresenceWriter();
+                        }
+                    }
+                }
+                return instance;
+            }
+        }
+
+        private void StartTimer() {
```

Comment:
nit: { on newlines.  

Recommend using Resharper to reformat it according to https://github.com/Microsoft/xbox-live-unity-plugin/blob/master/xbox-live-unity-plugin.sln.DotSettings